File: python/modify_order.py
"""
Modify Order Example

Place a resting order and then modify its price.
"""

from hyperliquid_sdk import HyperliquidSDK

sdk = HyperliquidSDK()

# Place a resting order
mid = sdk.get_mid("BTC")
limit_price = int(mid * 0.97)
order = sdk.buy("BTC", notional=11, price=limit_price, tif="gtc")
print(f"Placed order at ${limit_price:,}")
print(f"  OID: {order.oid}")

# Modify to a new price (4% below mid)
new_price = int(mid * 0.96)
new_order = order.modify(price=new_price)
print(f"Modified to ${new_price:,}")
print(f"  New OID: {new_order.oid}")

# Clean up
new_order.cancel()
print("Order cancelled.")

File: typescript/modify_order.ts
#!/usr/bin/env npx ts-node
/**
 * Modify Order Example
 *
 * Place a resting order and then modify its price.
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

async function main() {
  const sdk = new HyperliquidSDK();

  // Place a resting order
  const mid = await sdk.getMid("BTC");
  const limitPrice = Math.floor(mid * 0.97);
  const order = await sdk.buy("BTC", { notional: 11, price: limitPrice, tif: "gtc" });
  console.log(`Placed order at $${limitPrice.toLocaleString()}`);
  console.log(`  OID: ${order.oid}`);

  // Modify to a new price (4% below mid)
  const newPrice = Math.floor(mid * 0.96);
  const newOrder = await order.modify({ price: newPrice });
  console.log(`Modified to $${newPrice.toLocaleString()}`);
  console.log(`  New OID: ${newOrder.oid}`);

  // Clean up
  await newOrder.cancel();
  console.log("Order cancelled.");
}

main().catch(console.error);
